okio/jvm/src/main/java/okio/Buffer.kt

```diff
@@ -1299,7 +1280,7 @@ class Buffer : BufferedSource, BufferedSink, Cloneable, ByteChannel {
       var offset = 0L
       while (true) {
         val nextOffset = offset + (s.limit - s.pos)
-        if (nextOffset >= fromIndex) break
```

Comment:
this might have been a bug in `indexOf()`. I’m looking.

Reply:
It was a performance bug, not a correctness bug! We were returning one segment too soon in some off-by-one situations.